PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs

```diff
@@ -482,18 +570,31 @@ public Collection<Variable> GetPSObjectVariable(string varName)
         public IEnumerable<CallStack> GetCallStack()
         {
             ServiceCommon.Log("Obtaining the context for wcf callback");
-            List<CallStackFrame> callStackFrames = new List<CallStackFrame>();
+            List<CallStack> callStackFrames = new List<CallStack>();
 
             foreach (var psobj in _callstack)
             {
-                var frame = psobj.BaseObject as CallStackFrame;
-                if (frame != null)
+                if (_runspace.ConnectionInfo == null)
                 {
-                    callStackFrames.Add(frame);
+                    var frame = psobj.BaseObject as CallStackFrame;
+                    if (frame != null)
+                    {
+                        callStackFrames.Add(new CallStack(frame.ScriptName, frame.FunctionName, frame.ScriptLineNumber));
+                    }
+                }
+                else
+                {
+                    dynamic psFrame = (dynamic)psobj;

```

Comment:
type expected here?  It looks like we're expecting whatever psFrame is to have some properties defined